src/arch/windows.c

```diff
@@ -0,0 +1,27 @@
+#include <windows.h>
+
+void __stacker_black_box() {}
+
+PVOID __stacker_get_current_fiber() {
+    return GetCurrentFiber();
+}
+
+static size_t calc_stack_limit(size_t stack_low, size_t stack_guarantee) {
+    return stack_low +
+           max(stack_guarantee, sizeof(void *) == 4 ? 0x1000 : 0x2000) + // The guaranteed pages on a stack overflow 
+           0x1000; // The guard page
+}
+
+#if defined(_M_X64)
+size_t __stacker_get_stack_limit() {

```

Comment:
I think it'd be good to have some comments around here too, I'm not really sure what all these magical numbers are or where they come from.

Reply:
This is basically just a fast path for x86. The number are from Windows libraries and are offsets into the [Thread Environment Block](https://en.wikipedia.org/wiki/Win32_Thread_Information_Block). The other path uses `VirtualQuery` (a syscall) to get the stack size.

Reply:
We could just drop support for extending stacks of Windows fibers, we don't support them on Unix platforms either. Instead we could just cache the value in TLS.

We could also store the stack bounds in TLS and query them again if we find ourself on a new stack. That may also work on Unix platforms. Checking the stack bounds is a good sanity check to ensure that we won't unnecessarily expand the stack if we leave the original stack.

Reply:
I'm not sure I understand. What value should we cache?

By "expand the stack" you mean add another stack allocation or actually expand the existing allocation?